### PR TITLE
Minor: Add KIP-923 to upgrade-guide.html and dsl-api.html

### PR DESCRIPTION
**Pull request by wcarlson5**
add stream table join grace period to the docs.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: docs/streams/developer-guide/dsl-api.html [hunk 1]
```diff
@@ -2849,6 +2851,12 @@ <h5><a class="toc-backref" href="#id34">KTable-KTable Foreign-Key
                                         <li>When the table is <a class="reference internal" href="#versioned-state-stores"><span class="std std-ref">versioned</span></a>,
                                             the table record to join with is determined by performing a timestamped lookup, i.e., the table record which is joined will be the latest-by-timestamp record with timestamp
                                             less than or equal to the stream record timestamp. If the stream record timestamp is older than the table's history retention, then the record is dropped.</li>
+                                        <li>To use the Grace Period, the table needs to be <a class="reference internal" href="#versioned-state-stores"><span class="std std-ref">versioned</span></a>.
+                                            This will cause the stream to buffer for that duration before trying to find a matching record with the right timestamp in the table.
```
Comment (mjsax) [inline docs/streams/developer-guide/dsl-api.html]: > for that duration

unclear reference -> `for the specified grace period` (or something like this?)

File: docs/streams/developer-guide/dsl-api.html [hunk 2]
```diff
@@ -2849,6 +2851,12 @@ <h5><a class="toc-backref" href="#id34">KTable-KTable Foreign-Key
                                         <li>When the table is <a class="reference internal" href="#versioned-state-stores"><span class="std std-ref">versioned</span></a>,
                                             the table record to join with is determined by performing a timestamped lookup, i.e., the table record which is joined will be the latest-by-timestamp record with timestamp
                                             less than or equal to the stream record timestamp. If the stream record timestamp is older than the table's history retention, then the record is dropped.</li>
+                                        <li>To use the Grace Period, the table needs to be <a class="reference internal" href="#versioned-state-stores"><span class="std std-ref">versioned</span></a>.
```
Comment (mjsax) [inline docs/streams/developer-guide/dsl-api.html]: `[g]race [p]eriod`

not need to capitalize?

File: docs/streams/upgrade-guide.html [hunk 3]
```diff
@@ -198,6 +198,16 @@ <h3><a id="streams_api_changes_360" href="#streams_api_changes_360">Streams API
...
     </pre>
     </p>
+
+    <p>
+        KStreams-to-KTable joins now have an option for adding a grace period.
+        The grace period is enabled on the <code>Joined</code> object using with <code>withGracePeriod()</code> method.
+        This change was introduced in <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-923%3A+Add+A+Grace+Period+to+Stream+Table+Join">KIP-923</a>.
+        To use the grace period option in the Stream-Table join the table must be versioned, see the
+        <a href="/{{version}}/documentation/streams/developer-guide/dsl-api.html#versioned-state-stores">Developer Guide</a>.
+        For more information, including how it can be enabled and further configured, see the <a href="/{{version}}/documentation/streams/developer-guide/dsl-api.html#kstream-ktable-join"><b>Kafka Streams Developer Guide</b></a>.
```
Comment (mjsax) [inline docs/streams/upgrade-guide.html]:
```
the table must be versioned, see the
        <a href="/{{version}}/documentation/streams/developer-guide/dsl-api.html#versioned-state-stores">Developer Guide</a>.        
```

Use `versioned` as link and remove the `, see the Developer Guide` part.

(This change also avoid the usage of `Developer Guide` in a similar way in two consecutive sentences.

**Review by mjsax — COMMENTED** [review]
Thanks! -- Couple of suggestions.

File: docs/streams/developer-guide/dsl-api.html (hunk 2 above)
Comment (wcarlson5) [reply docs/streams/developer-guide/dsl-api.html]: I could go either way. Lets un cap it for now

**Comment by wcarlson5** [comment]
@mjsax Finally got around to your comments. I think we need to pick this back to 3.6 and 3.7 as well

**Comment by mjsax** [comment]
Merged to `trunk` and cherry-picked to `3.7` and `3.6` branches. -- @wcarlson5 can you do a PR on `kafka-site` to get the change live for `3.6`?

**Pull request merged**